plugins/net_plugin/net_plugin.cpp

```diff
@@ -1838,6 +1838,25 @@ namespace eosio {
          return send_buffer;
       }
 
+      static send_buffer_type create_send_buffer_from_serialized_block( const std::vector<char>& v ) {
+         static_assert( signed_block_which == fc::get_index<net_message, signed_block>() );
+
+         // match net_message static_variant pack
+         const uint32_t which_size = fc::raw::pack_size( unsigned_int( signed_block_which ) );
+         const uint32_t payload_size = which_size + v.size();
+
+         const char* const header = reinterpret_cast<const char* const>(&payload_size); // avoid variable size encoding of uint32_t
+         const size_t buffer_size = message_header_size + payload_size;
+
+         auto send_buffer = std::make_shared<vector<char>>( buffer_size );
+         fc::datastream<char*> ds( send_buffer->data(), buffer_size );
+         ds.write( header, message_header_size );
+         fc::raw::pack( ds, unsigned_int( signed_block_which ) );
+         ds.write( v.data(), v.size() );
```

Comment:
Instead of `vector<char> read_serialized_block_by_num(block_num)` how about `void read_serialized_block_by_num(block_num, vector<char>& data)`. 

`read_serialized_block_by_num` would append to the `data` buffer. We can then avoid the copy here.